Updating the Postgres Documentation (#28119)
DataDog/documentation: layouts/shortcodes/dbm/postgres/self-hosted-install-kubernetes.en.md

> If you have a Kubernetes cluster, use the [Datadog Cluster Agent][1] for Database Monitoring.

Reviewer: ```suggestion
If you're running a Kubernetes cluster, use the [Datadog Cluster Agent][1] to enable Database Monitoring.
```

> Follow the instructions to [enable the cluster checks][2] if not already enabled for your Datadog Cluster Agent.

Reviewer: ```suggestion
**Note**: Make sure [cluster checks][2] are enabled for your Datadog Cluster Agent before proceeding.
```

> Below are some examples of how you can configure the Postgres configuration for different Datadog Cluster Agent deployment methods:

Reviewer: ```suggestion
Below are step-by-step instructions for configuring the Postgres integration using different Datadog Cluster Agent deployment methods.
```

### Operator

> Using [Kubernetes and Integration | Operator][3] as a reference, below is how you would setup the Postgres Integration:

Reviewer: ```suggestion
Using the [Operator instructions in Kubernetes and Integrations][3] as a reference, follow the steps below to set up the Postgres integration:
```

> 1. Use the example below to create or update the `datadog-agent.yaml`:

Reviewer: ```suggestion
1. Create or update the `datadog-agent.yaml` file with the following configuration:
```

    ```yaml
    apiVersion: datadoghq.com/v2alpha1
    kind: DatadogAgent
    metadata:
      name: datadog
    spec:
      global:
        clusterName: <CLUSTER_NAME>
        site: <DD_SITE>
        credentials:
          apiSecret:
            secretName: datadog-agent-secret
            keyName: api-key

      features:
        clusterChecks:
          enabled: true

      override:
        nodeAgent:
          image:
            name: agent
            tag: 7.63.3

        clusterAgent:
          extraConfd:
            configDataMap:
              postgres.yaml: |-
                cluster_check: true
                init_config:
                instances:
                - host: <HOST>
                  port: 5432
                  username: datadog
                  password: 'ENC[datadog_user_database_password]'
                  dbm: true
    ```

>     **Note**: For Postgres 9.6, add the following settings to the instance config where host and port are specified:

Reviewer: ```suggestion
    **Note**: For Postgres 9.6, add the following lines to the instance config where host and port are specified:
```

    ```yaml
    pg_stat_statements_view: datadog.pg_stat_statements()
    pg_stat_activity_view: datadog.pg_stat_activity()
    ```

> 2. Apply the changes to the Datadog Operator with the following command:

Reviewer: ```suggestion
2. Apply the changes to the Datadog Operator using the following command:
```

    ```shell
    kubectl apply -f datadog-agent.yaml
    ```
  
### Helm

> Using [Kubernetes and Integration | Helm][4] as a reference, below is how you would setup the Postgres Integration:

Reviewer: ```suggestion
Using the [Helm instructions in Kubernetes and Integrations][4] as a reference, follow the steps below to set up the Postgres integration:
```

> 1. Update your YAML configuration file (`datadog-values.yaml` in the Cluster Agent installation instructions) to include the following:

Reviewer: ```suggestion
1. Update your `datadog-values.yaml` file (used in the Cluster Agent installation instructions) with the following configuration:
```

    ```yaml
    datadog:
      clusterChecks:
        enabled: true

    clusterChecksRunner:
      enabled: true

    clusterAgent:
      enabled: true
      confd:
        postgres.yaml: |-
          cluster_check: true
          init_config:
          instances:
          - dbm: true
            host: <HOST>
            port: 5432
            username: datadog
            password: 'ENC[datadog_user_database_password]'

    ```

>     **Note**: For Postgres 9.6, add the following settings to the instance config where host and port are specified:

Reviewer: ```suggestion
    **Note**: For Postgres 9.6, add the following lines to the instance config where host and port are specified:
```

    ```yaml
    pg_stat_statements_view: datadog.pg_stat_statements()
    pg_stat_activity_view: datadog.pg_stat_activity()
    ```

> 2. Deploy the Agent with the above configuration file from the command line:

Reviewer: ```suggestion
2. Deploy the Agent with the above configuration file using the following command:
```

    ```shell
    helm install datadog-agent -f datadog-values.yaml datadog/datadog
    ```

<div class="alert alert-info">
For Windows, append <code>--set targetSystem=windows</code> to the <code>helm install</code> command.
</div>

### Configure with mounted files

> To configure a cluster check with a mounted configuration file, mount the configuration file in the Cluster Agent container at the path: `/conf.d/postgres.yaml`:

Reviewer: ```suggestion
To configure a cluster check with a mounted configuration file, mount the configuration file in the Cluster Agent container at the `/conf.d/postgres.yaml`:
```

Author: Feels kinda weird to not say "path" or "location" before the location of the file.

```yaml
cluster_check: true  # Make sure to include this flag
init_config:
instances:
  - dbm: true
    host: '<HOST>'
    port: 5432
    username: datadog
    password: 'ENC[datadog_user_database_password]'

    ## Required: For Postgres 9.6, uncomment these lines to use the functions created in the setup
    # pg_stat_statements_view: datadog.pg_stat_statements()
    # pg_stat_activity_view: datadog.pg_stat_activity()
```

### Configure with Kubernetes service annotations

> Rather than mounting a file, you can declare the instance configuration as a Kubernetes Service. To configure this check for an Agent running on Kubernetes, create a Service in the same namespace as the Datadog Cluster Agent:

Reviewer: ```suggestion
Instead of mounting a file, you can declare the instance configuration as a Kubernetes Service. To configure this check for an Agent running on Kubernetes, create a Service in the same namespace as the Datadog Cluster Agent:
```

#### Autodiscovery Annotations v2

```yaml
apiVersion: v1
kind: Service
metadata:
  name: postgres
  labels:
    tags.datadoghq.com/env: '<ENV>'
    tags.datadoghq.com/service: '<SERVICE>'
  annotations:
    ad.datadoghq.com/<CONTAINER_NAME>.checks: |
      {
        "postgres": {
          "init_config": <INIT_CONFIG>,
          "instances": [
            {
              "dbm": true,
              "host": "<HOST>",
              "port": 5432,
              "username": "datadog",
              "password": "ENC[datadog_user_database_password]"
            }
          ]
        }
      }      
spec:
  ports:
  - port: 5432
    protocol: TCP
    targetPort: 5432
    name: postgres
```

> Visit [Autodiscovery Annotations][5] for more information.

Reviewer: ```suggestion
For more information, see [Autodiscovery Annotations][5].
```

> For Postgres 9.6, add the following settings to the instance config where host and port are specified:

Reviewer: ```suggestion
If you're using Postgres 9.6, add the following to the instance configuration:
```

```json
"pg_stat_statements_view": "datadog.pg_stat_statements()",
"pg_stat_activity_view": "datadog.pg_stat_activity()"
```

> The Cluster Agent automatically registers this configuration and begin running the Postgres check.

Reviewer: ```suggestion
The Cluster Agent automatically registers this configuration and begins running the Postgres check.
```
